### PR TITLE
[feature](function) support ip function named ipv6_cidr_to_range(addr, cidr)

### LINKED ISSUE #20584: [Enhancement] Add IP address process function for SQL like ClickHouse
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

ClickHouse support some [IP address process function](https://clickhouse.com/docs/zh/sql-reference/functions/ip-address-functions) for SQL. These are very useful for some network service cases.
Hope to add these function ASAP.Thanks.

### Solution

Add IP address process function for SQL like ClickHouse

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sjyango**
## Proposed changes

Issue Number: close #20584

Accepts an IPv6 and a Int value containing the CIDR. Return a struct with two IPv6 containing the lower range(named `min`) and the higher range(named `max`) of the subnet.

Doris output:
image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sjyango** [comment]
run buildall

**Comment by sjyango** [comment]
run pipeline_x

**Comment by sjyango** [comment]
run pipelinex_p0

File: be/src/vec/functions/function_ip.h [hunk 1]
```diff
@@ -832,4 +835,128 @@ class FunctionIsIPAddressInRange : public IFunction {
...
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        ColumnPtr& addr_column = block.get_by_position(arguments[0]).column;
+        ColumnPtr& cidr_column = block.get_by_position(arguments[1]).column;
+        const IColumn* cidr_col = nullptr;
+        const NullMap* cidr_nullmap = nullptr;
+        ColumnPtr col_res = nullptr;
+
+        if (cidr_column->is_nullable()) {
```
Comment (amorynan) [inline be/src/vec/functions/function_ip.h:876]: do not use columntype to judge what type of current data, use block.get_data_type

File: be/src/vec/functions/function_ip.h [hunk 2]
```diff
@@ -832,4 +835,128 @@ class FunctionIsIPAddressInRange : public IFunction {
...
+            cidr_nullmap = &cidr_column_nullable->get_null_map_data();
+        } else {
+            cidr_col = cidr_column.get();
+        }
+
+        if (addr_column->is_nullable()) {
+            const auto* addr_column_nullable =
+                    assert_cast<const ColumnNullable*>(addr_column.get());
+            const NullMap* addr_nullmap = &addr_column_nullable->get_null_map_data();
+            if (const auto* ipv6_addr_column = check_and_get_column<ColumnIPv6>(
```
Comment (amorynan) [inline be/src/vec/functions/function_ip.h:889]: same with is_null
use data type to check

File: be/src/vec/functions/function_ip.h [hunk 3]
```diff
@@ -832,4 +835,128 @@ class FunctionIsIPAddressInRange : public IFunction {
...
+        const NullMap* cidr_nullmap = nullptr;
+        ColumnPtr col_res = nullptr;
+
+        if (cidr_column->is_nullable()) {
+            const auto* cidr_column_nullable =
+                    assert_cast<const ColumnNullable*>(cidr_column.get());
+            cidr_col = &cidr_column_nullable->get_nested_column();
+            cidr_nullmap = &cidr_column_nullable->get_null_map_data();
+        } else {
+            cidr_col = cidr_column.get();
```
Comment (amorynan) [inline be/src/vec/functions/function_ip.h:882]: if cidr column is column vector , just assert cast. do not use IColumn.

File: be/src/vec/functions/function_ip.h [hunk 4]
```diff
@@ -832,4 +835,128 @@ class FunctionIsIPAddressInRange : public IFunction {
...
+        auto& vec_res_null_map = col_res_null_map->get_data();
+
+        static constexpr Int64 max_cidr_mask = IPV6_BINARY_LENGTH * 8;
+
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            if ((from_nullmap && (*from_nullmap)[i]) || (cidr_nullmap && (*cidr_nullmap)[i])) {
+                vec_res_null_map[i] = 1;
+                continue;
+            }
+            Int64 cidr = cidr_column.get_int(i);
```
Comment (amorynan) [inline be/src/vec/functions/function_ip.h:938]: make function param cidr column is column vector

**Comment by sjyango** [comment]
run buildall

**Comment by sjyango** [comment]
run performance

**Review by amorynan — APPROVED** [review]
LGTM

**Review by starocean999 — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Comment by HappenLee** [comment]
@sjyango Please add a doc for the function

**Comment by HappenLee** [comment]
the function seems better be depend on argument?

**Pull request merged**